### ISSUE TITLE
Possible to attachRole using role name

### ISSUE THREAD

**Issue by imujjwal96** [Zizaco/entrust#816]
I am receiving role name from the front end and would like to attach role to a user while registering. 
When i executed this
```         
$user->attachRole('doctor'); // doctor is a role name
```
Got the error
![image](https://user-images.githubusercontent.com/8245662/28638136-8fa6bdfe-7261-11e7-84cc-7e11d483fcb7.png)

Since the role name is also unique, shouldn't it be possible to attachRole using role name?

**Comment by Radiergummi** [comment]
You could just use a `<select>` with the option values as IDs instead of names - that's how I solved it at least...

**Comment by imujjwal96** [comment]
That's how i went with it! Would it not be possible to have the as name a possible input?

**Comment by Radiergummi** [comment]
Yeah, but this project seems to be pretty abandoned. I ran into more problems while writing the `update` method - need to check whether the role has already been set, but since we changed the chosen role to be an ID, we can't directly use `hasRole` anymore, since that one only accepts a string. Eventually I went with

````
if ( ! $user->hasRole( Role::find( $request->role )->name ) ) {
            $user->attachRole( $request->role );
        }
````

...but that's ugly as hell, and a lot of unnecessary queries too. I really don't know what the original line of thinking was when this library was built.
